lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4787,6 +4787,34 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           }
         }
 
+        // Require the superclass to have an available designated initializer,
+        // unless we have already determined the class or superclass is invalid.
+        if (!isInvalidSuperclass && !Super->isInvalid()) {
+          bool superIsConstructible = false; // can be satisfied by convenience initializers
+          bool superHasDesignatedInitializer = false; // is subclassable
+          for (auto member : Super->getMembers()) {
```

Comment:
[Turn predicate loops into predicate functions?](https://llvm.org/docs/CodingStandards.html#turn-predicate-loops-into-predicate-functions)

Reply:
Might work. I'd like to find a refactor that cleans this up for sure.